### PR TITLE
Implement the sqrt intrinsic

### PR DESCRIPTION
**Pull request by AndyAyersMS**

### CODE REVIEW

File: include/Reader/readerir.h [hunk 1]
```diff
@@ -459,9 +459,12 @@ class GenIR : public ReaderBase {
...
   IRNode *stringGetChar(IRNode *Arg1, IRNode *Arg2) override;
 
-  bool sqrt(IRNode *Arg1, IRNode **RetVal) override {
-    throw NotYetImplementedException("sqrt");
-  };
+  /// Optionally generate inline code for the \p sqrt operation
+  ///
+  /// \param Argument      input value for sqrt
+  /// \param Result [out]  resulting sqrt value, if reader decided to expand
+  /// \returns             true if Result represents the sqrt
```
Comment (JosephTremoulet) [inline include/Reader/readerir.h]: iff?

File: lib/Reader/readerir.cpp [hunk 2]
```diff
@@ -5479,6 +5479,22 @@ bool GenIR::abs(IRNode *Argument, IRNode **Result) {
   return false;
 }
 
+bool GenIR::sqrt(IRNode *Argument, IRNode **Result) {
+  Type *Ty = Argument->getType();
+
+  if (Ty->isFloatingPointTy()) {
+    Type *Types[] = {Ty};
+    Value *FSqrt = Intrinsic::getDeclaration(JitContext->CurrentModule,
+                                             Intrinsic::sqrt, Types);
```
Comment (JosephTremoulet) [inline lib/Reader/readerir.cpp]: Can we lower this or does it expect a helper we don't have?

**Comment by JosephTremoulet** [comment]
LGTM aside from one nit and one question.

File: lib/Reader/readerir.cpp (hunk 2 above)
Comment (AndyAyersMS) [inline lib/Reader/readerir.cpp]: It lowers fine on x64. Not sure if we'll hit issues with other platforms.

File: include/Reader/readerir.h (hunk 1 above)
Comment (AndyAyersMS) [inline include/Reader/readerir.h]: Sure. Also this should be in reader.h, not readerir.h, will move.

File: lib/Reader/readerir.cpp (hunk 2 above)
Comment (JosephTremoulet) [inline lib/Reader/readerir.cpp]: Ok.  Thanks for verifying.

I assume that if we do hit issues with other platforms we'll want to use this intrinsic and add the right lowering under our environment anyway.

**Comment by AndyAyersMS** [comment]
Latest commit was just a squash so am going to merge.

**Pull request merged**